File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        map={}
        for i,n in enumerate(nums):
            ntf=target-n
            if ntf in map and map[ntf]!=i:
                return [i, map[ntf]]
            else:
                map[n]=i




            

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {

    dic={}

    for (let x=0; x<nums.length; x++){
        let need=target-nums[x]
        if (need in dic) {
            return [dic[need], x];
        } else{
            dic[nums[x]]=x
        }
    }




    
};